music21/musicxml/m21ToXml.py

```diff
@@ -3369,8 +3435,28 @@ def parseOneElement(self, obj):
             if parsedObject is False:
                 environLocal.printDebug(['did not convert object', obj])
 
-        for sp in postList:  # directions that follow the element
-            root.append(sp)
+        if not skipPrePostObjectSpanners:
+            for sp in postList:  # directions that follow the element
+                root.append(sp)
+
+    def _hasPrePostObjectSpanners(self, obj) -> bool:
+        '''
+        returns True if and only if:
+        (1) there are spanners related to the object that should appear before the object
+        to the <measure> tag, or (2) there are spanners related to the object that should
+        appear after the object in the measure tag.
+        '''
+        spannerBundle = self.spannerBundle.getBySpannedElement(obj)
+        if not spannerBundle:
+            return False
+
+        # this list of spanner classes must match the paramsSet keys in prePostObjectSpanners()
+        for m21spannerClass in ('Ottava', 'DynamicWedge', 'Line'):
+            for thisSpanner in spannerBundle.getByClass(m21spannerClass):
+                if thisSpanner.isFirst(obj) or thisSpanner.isLast(obj):
+                    return True
+
+        return False
 
     def prePostObjectSpanners(self, target):
```

Comment:
rename to `relatedSpanners(self, target: base.Music21Object) -> tuple[Sequence[Element], Sequence[Element]`  THANKS!

Reply:
done